packages/scene-composer/src/components/StateManager.tsx

```diff
@@ -148,6 +151,32 @@ const StateManager: React.FC<SceneComposerInternalProps> = ({
     }
   }, [activeCamera, selectedDataBinding]);
 
+  const getBindingsFromTag = (component: IAnchorComponentInternal): ITagData => {
+    return {
+      chosenColor: component.chosenColor,
+      navLink: component.navLink,
+      dataBindingContext: !component.valueDataBinding?.dataBindingContext
+        ? undefined
+        : applyDataBindingTemplate(component.valueDataBinding, dataBindingTemplate),
+    };
+  };
+  const getBindingsFromEntityBinding = (component: IEntityBindingComponentInternal): IEntityBindingInfo => {
+    return {
+      dataBindingContext: !component?.valueDataBinding?.dataBindingContext
+        ? undefined
+        : component?.valueDataBinding.dataBindingContext,
+    };
+  };
+  const getBindingsFromDataOverlay = (component: IDataOverlayComponentInternal): IDataOverlayInfo => {
+    const dataBindingContexts: unknown[] = [];
+    component.valueDataBindings.forEach((item) => {
+      if (item.valueDataBinding) {
+        dataBindingContexts.push(applyDataBindingTemplate(item.valueDataBinding, dataBindingTemplate));
+      }
+    });
+    return { dataBindingContexts };
+  };
```

Comment:
useCallback?

Reply:
moved to static util function library instead